x/finality/keeper/power_dist_change.go

```diff
@@ -320,6 +323,11 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 	// sort new finality providers in activeBTCDels to ensure determinism
 	fpActiveBtcPkHexList := make([]string, 0, len(activedSatsByFpBtcPk))
 	for fpBTCPKHex := range activedSatsByFpBtcPk {
+		// if the fp was slashed, should not even be added to the list
+		_, isSlashed := slashedFPs[fpBTCPKHex]
```

Comment:
hmm I think we should add it the active list and and later replicate the logic from above loop i.e 
```
		if isSlashed {
			fp.IsSlashed = true
			continue
		}
```

as otherwise we won't set jailed flag for new slashed fp. (not sure this is criticial, but this is definitly inconsisent)

Reply:
good point